packages/eslint-rules/src/rules/propagate_error_cause.ts

```diff
@@ -0,0 +1,191 @@
+import { ESLintUtils } from '@typescript-eslint/utils';
+
+export const propagateErrorCause = ESLintUtils.RuleCreator.withoutDocs({
+  create(context) {
+    return {
+      // This naming comes from @typescript-eslint/utils types.
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      CatchClause(node) {
+        const checkNode = (errorType: string) => {
+          if (!node.param) {
+            // we encounter this, presumably the error will not be wrapped in If/SwitchStatements
+            for (const throwBody of node.body.body) {
+              if (
+                throwBody.type === 'ThrowStatement' &&
+                //@ts-expect-error assumes incorrect type for body.argument
+                throwBody.argument.callee.name === errorType
+              ) {
+                context.report({
+                  messageId: 'noCausePropagation',
+                  node,
+                });
+              }
+            }
+          } else if (
+            node.param &&
+            node.param.type === 'Identifier' &&
+            node.param.name &&
+            node.body.body.length
+          ) {
+            const causeVarNames = [];
+            causeVarNames.push(node.param.name);
+            const queue = [];
+            let body = node.body.body[0];
+
+            if (
+              body.type !== 'ThrowStatement' &&
+              body.type !== 'IfStatement' &&
+              body.type !== 'SwitchStatement' &&
+              body.type !== 'BlockStatement'
+            ) {
+              for (const newBody of node.body.body) {
+                if (newBody.type === 'VariableDeclaration') {
+                  //@ts-expect-error assumes incorrect type for body.declarations[0].id
```

Comment:
instead of `@ts-expect-error` you can `(body.declarations[0].id as <desiredType>).name`
this would be preferable way of doing this. See example here https://github.com/aws-amplify/amplify-backend/blob/040272d7d05427d2de82658e8de2ec0d60981b46/scripts/components/api-changes-validator/api_usage_statements_generators.ts#L225

The reason is that casting with `as` dismisses less potential problems than `@ts-expect-error`. We should use `@ts-expect-error` only if we exhausted other options.

Please apply this to other places.